qt/widgets/instrumentview/src/PanelsSurface.cpp

```diff
@@ -316,11 +317,11 @@ void PanelsSurface::addFlatBankOfDetectors(
   QVector<QPointF> vert;
   vert << p1 << p0;
   info->polygon = QPolygonF(vert);
-#pragma omp parallel for
   for (int i = 0; i < static_cast<int>(detectors.size()); ++i) {
     auto detector = detectors[i];
     addDetector(detector, pos0, index, info->rotation);
     UnwrappedDetector &udet = m_unwrappedDetectors[detector];
+    // this isn't thread-safe
```

Comment:
Is it possible/beneficial to use `#pragma omp ordered` around `info->polygon << QPointF(udet.u, udet.v);` so that the first part of the function can still be parallelised?